src/openvic-simulation/history/CountryHistory.cpp

```diff
@@ -0,0 +1,308 @@
+#include "CountryHistory.hpp"
+
+#include "openvic-simulation/GameManager.hpp"
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+CountryHistory::CountryHistory(
+	Culture const* new_primary_culture,
+	const std::vector<Culture const*>&& new_accepted_cultures,
+	Religion const* new_religion,
+	CountryParty const* new_ruling_party,
+	const Date new_last_election,
+	const std::map<Ideology const*, const fixed_point_t>&& new_upper_house,
+	Province const* new_capital,
+	GovernmentType const* new_government_type,
+	const fixed_point_t new_plurality,
+	NationalValue const* new_national_value,
+	const bool new_civilised,
+	const fixed_point_t new_prestige,
+	const std::vector<Reform const*>&& new_reforms,
+	OOB const* new_inital_oob
+) : primary_culture { new_primary_culture }, accepted_cultures { std::move(new_accepted_cultures) }, religion { new_religion }, ruling_party { new_ruling_party }, last_election { new_last_election }, upper_house { std::move(new_upper_house) }, capital { new_capital }, government_type { new_government_type }, plurality { new_plurality }, national_value { new_national_value }, civilised { new_civilised }, prestige { new_prestige }, reforms { std::move(new_reforms) }, inital_oob { new_inital_oob } {}
+
+Culture const* CountryHistory::get_primary_culture() const {
+	return primary_culture;
+}
+
+const std::vector<Culture const*>& CountryHistory::get_accepted_cultures() const {
+	return accepted_cultures;
+}
+
+Religion const* CountryHistory::get_religion() const {
+	return religion;
+}
+
+CountryParty const* CountryHistory::get_ruling_party() const {
+	return ruling_party;
+}
+
+const Date CountryHistory::get_last_election() const {
+	return last_election;
+}
+
+const std::map<Ideology const*, const fixed_point_t>& CountryHistory::get_upper_house() const {
+	return upper_house;
+}
+
+Province const* CountryHistory::get_capital() const {
+	return capital;
+}
+
+GovernmentType const* CountryHistory::get_government_type() const {
+	return government_type;
+}
+
+const fixed_point_t CountryHistory::get_plurality() const {
+	return plurality;
+}
+
+NationalValue const* CountryHistory::get_national_value() const {
+	return national_value;
+}
+
+const bool CountryHistory::is_civilised() const {
+	return civilised;
+}
+
+const fixed_point_t CountryHistory::get_prestige() const {
+	return prestige;
+}
+
+const std::vector<Reform const*>& CountryHistory::get_reforms() const {
+	return reforms;
+}
+
+OOB const* CountryHistory::get_inital_oob() const {
+	return inital_oob;
+}
+
+#define EXISTING_ENTRY country_histories.at(country).at(date)
```

Comment:
Should really undef this so its limiting its scope, also since its only ever used for `add_country_history_entry`, should really keep it inside that function's block, especially since it uses the arguments for it, it makes it easier to see what its relevant to.

Though I have to wonder why not just save the iterator once and reference the iterator instead?